src/components/calcite-shell-center-row/calcite-shell-center-row.tsx

```diff
@@ -33,19 +35,50 @@ export class CalciteShellCenterRow {
    */
   @Prop({ reflect: true }) position: CalcitePosition = "end";
 
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteShellCenterRowElement;
+
   // --------------------------------------------------------------------------
   //
   //  Render Methods
   //
   // --------------------------------------------------------------------------
 
   render(): VNode {
-    return (
-      <Host>
-        <div class={CSS.content}>
-          <slot />
-        </div>
-      </Host>
+    const { el } = this;
+    const rtl = getElementDir(el) === "rtl";
+    
+    const contentNode = (
+      <div class={{ [CSS.content]: true, [CSS_UTILITY.rtl]: rtl }}>
+        <slot />
+      </div>
     );
+
+    const actionBar = getSlotted<HTMLCalciteActionBarElement>(el, SLOTS.actionBar);
+
+    if (!actionBar) {
+      return <Host>{contentNode}</Host>;
+    }
+
+    const actionBarNode = (
+      <div class={{ [CSS.actionBarContainer]: true, [CSS_UTILITY.rtl]: rtl }}>
+        <slot name={SLOTS.actionBar} />
+      </div>
+    );
+
+    const children = [contentNode];
```

Comment:
suggestion: 

```tsx
const children = [actionBarNode, contentNode];

if(actionBar.position === "end"){
children.reverse();
}

```

Reply:
Wouldn't I need to nest this in a conditional for whether or not there's an action-bar?

Reply:
Oh...I see. Lemme try dat.

Reply:
I don't think so. it won't render anything null